Sources/CoreMetrics/Metrics.swift

```diff
@@ -12,220 +12,230 @@
 //
 //===----------------------------------------------------------------------===//
 
-public protocol Counter: AnyObject {
+public protocol CounterHandler: AnyObject {
     func increment<DataType: BinaryInteger>(_ value: DataType)
 }
 
-public extension Counter {
+public class Counter: CounterHandler {
+    @usableFromInline
+    var handler: CounterHandler
+    public let label: String
+    public let dimensions: [(String, String)]
+
+    // this method is public to provide an escape hatch for situations one must use a custom factory instead of the gloabl one
+    // we do not expect this API to be used in normal circumstances, so if you find yourself using it make sure its for a good reason
+    public init(label: String, dimensions: [(String, String)], handler: CounterHandler) {
```

Comment:
Yup, thanks; good to have as escape hatch... (more necessarily in the logging apis, but definitely good to have the same escape hatch mechanism in both)